src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victum) {
+    if (victum instanceof Herbivore && victum.hidden === false) {
+      victum.health -= 50;
+    }
+
+    if (victum.health === 0) {
+      Animal.alive.splice(Animal.alive.indexOf(victum), 1);
+    }
```

Comment:
<img width="666" alt="CleanShot 2022-10-04 at 13 57 03@2x" src="https://user-images.githubusercontent.com/103505670/193802118-bacc9939-c2dc-4c1e-b5d7-ad816ac8edcd.png">

try to use filter here

Reply:
Use filter here, also would be better to check if health is less or equal than 0 (in case if we want to change health or damage - it shouldn't require any refactoring)